### PR TITLE
Reordered ClickGUI, added InfoDisplay, added basic ElytraFly modes

### PR DESCRIPTION
**Pull request by tonio-cartonio**
I increased version number to `2.9.1`
I reordered the click GUI, now most mods aren't under `Player` anymore.
I added the possibility to hide certain mods from the ModList (not setting `hidden`!)
I added 2 modes to ElytraFlight : a basic `control` mode which won't keep altitude and `packet` mode (with infinite durability and nice speeds, but patched on 2b).
AntiBats is not AntiUselessMobs and can hide squids too.
ThisModIsDedicatedToUfoCrossing has now a name which won't take up all screen.
MatrixNotifications can now hook to Discord Webhooks (set `provider`) and can relay in-game chat completely.
I added a configurable watermark mod which by default shows ForgeHax version.
I added a configurable ChatWatermark (some people actually use it wtf?????)

OverFloyd added InfoDisplay with some system info (FPS, ping, Saturation...) and I'm PRing his changes too!

### CODE REVIEW

**Comment by 0-x-2-2** [comment]
MatrixNotifications should have always worked with discord you just had to use the slack compatibility discord webhook url

File: src/main/java/com/matt/forgehax/mods/ElytraFlight.java [hunk 1]
```diff
@@ -27,53 +34,183 @@
...
+          .name("maxboost")
+          .description("Max speed in control mode")
+          .defaultTo(1.60D)
+          .build();
+  
+  public final Setting<Double> up_speed =
+      getCommandStub()
+          .builders()
+          .<Double>newSettingBuilder()
+          .name("up_speed")
```
Comment (fr1kin) [inline src/main/java/com/matt/forgehax/mods/ElytraFlight.java:69]: Use `-`  over `_` in command names (it's okay to use it in the field name)

File: src/main/java/com/matt/forgehax/mods/MatrixNotifications.java [hunk 2]
```diff
@@ -113,7 +113,7 @@
           .<String>newSettingBuilder()
           .name("skin-server-url")
           .description("URL to the skin server. If left empty then no image will be used.")
-          .defaultTo("https://visage.surgeplay.com/face/160/")
+          .defaultTo("https://minotar.net/helm/")
```
Comment (fr1kin) [inline src/main/java/com/matt/forgehax/mods/MatrixNotifications.java:116]: Any specific reason for this change?

File: src/main/java/com/matt/forgehax/mods/MatrixNotifications.java [hunk 3]
```diff
@@ -151,9 +160,18 @@
           .description("Message when player moves in the queue")
           .defaultTo(true)
           .build();
+
+  private final Setting<String> provider =
```
Comment (fr1kin) [inline src/main/java/com/matt/forgehax/mods/MatrixNotifications.java:164]: Use an enum setting type here aswell

File: src/main/java/com/matt/forgehax/mods/MatrixNotifications.java [hunk 4]
```diff
@@ -224,13 +242,22 @@ private static String getUriUuid() {
   
   private void notify(String message) {
     JsonObject object = new JsonObject();
-    object.addProperty("text", message);
-    object.addProperty("format", "plain");
-    object.addProperty("displayName", MC.getSession().getUsername());
+	if (provider.get().equals("Discord")) {
+    	object.addProperty("content", message);
```
Comment (fr1kin) [inline src/main/java/com/matt/forgehax/mods/MatrixNotifications.java:246]: 😭 tabbing pls

File: src/main/java/com/matt/forgehax/util/mod/ToggleMod.java [hunk 5]
```diff
@@ -27,6 +28,22 @@ public ToggleMod(Category category, String modName, boolean defaultValue, String
...
             .build();
+    this.nolist =
+        getCommandStub()
+            .builders()
+            .<Boolean>newSettingBuilder()
+            .name("hidden")
+            .description("Hides the mod from modlist")
+            .defaultTo(defaultValue)
+            .changed(
+                cb -> {
```
Comment (fr1kin) [inline src/main/java/com/matt/forgehax/util/mod/ToggleMod.java:39]: This probably shouldn't be enabling/disabling the mod...

Comment (tonio-cartonio) [reply src/main/java/com/matt/forgehax/util/mod/ToggleMod.java]: Indeed, sorry! Fixed

File: src/main/java/com/matt/forgehax/mods/ElytraFlight.java [hunk 6]
```diff
@@ -27,53 +34,183 @@
...
+  public final Setting<Double> down_speed =
+      getCommandStub()
+          .builders()
+          .<Double>newSettingBuilder()
+          .name("down_speed")
+          .description("Downward speed")
           .defaultTo(0.05D)
           .build();
+
+  public final Setting<String> mode =
```
[the remark this answers is not in the document]
Comment (tonio-cartonio) [reply src/main/java/com/matt/forgehax/mods/ElytraFlight.java]: Enums are indeed nice! Switched to them

File: src/main/java/com/matt/forgehax/mods/ElytraFlight.java [hunk 7]
```diff
@@ -2,19 +2,26 @@
...
+import net.minecraft.network.play.client.CPacketPlayer;
+import net.minecraft.network.play.server.SPacketEntityMetadata;
+import net.minecraft.network.play.server.SPacketPlayerPosLook;
 import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
 import net.minecraftforge.fml.relauncher.Side;
 import net.minecraftforge.fml.relauncher.SideOnly;
+import net.minecraftforge.fml.common.ObfuscationReflectionHelper;
 
 @RegisterMod
 public class ElytraFlight extends ToggleMod {
```
[the remark this answers is not in the document]
Comment (tonio-cartonio) [reply src/main/java/com/matt/forgehax/mods/ElytraFlight.java:24]: Should I copy over the ControlMode from your 1.15 branch? I am using a more edited ControlMode anyway, this was supposed to be a "basic fly" a-la-impact to have an option built-in the public ForgeHax

File: src/main/java/com/matt/forgehax/mods/ElytraFlight.java (hunk 1 above)
Comment (tonio-cartonio) [reply src/main/java/com/matt/forgehax/mods/ElytraFlight.java]: Nice! Will change it also in other places

File: src/main/java/com/matt/forgehax/mods/MatrixNotifications.java (hunk 2 above)
Comment (tonio-cartonio) [reply src/main/java/com/matt/forgehax/mods/MatrixNotifications.java:116]: It has a shorter url and looks nicer in chat on my super small screen :) I'll revert, pushed by mistake, was supposed to be a default change just for me

**Comment by tonio-cartonio** [comment]
> MatrixNotifications should have always worked with discord you just had to use the slack compatibility discord webhook url

I had no idea, how to get the slack compatibility webhook url tho?

Also @fr1kin since there is no need to change the format of the POST, should I drop all changes I made on MatrixNotifications (which mostly are just "adding support" for Discord and an extremely obnoxious chat relay)?

File: src/main/java/com/matt/forgehax/mods/MatrixNotifications.java (hunk 4 above)
Comment (tonio-cartonio) [reply src/main/java/com/matt/forgehax/mods/MatrixNotifications.java]: wow I don't know what happened here, sorry it slipped!

**Comment by tonio-cartonio** [comment]
Closed for different reasons, will redo

**Pull request closed without merging**